src/js/actions/layers.js

```diff
@@ -144,7 +146,19 @@ define(function (require, exports) {
         });
 
         return Promise.join(requiredPropertiesPromise, optionalPropertiesPromise, function (required, optional) {
-            return _.chain(required).zipWith(optional, _.merge).reverse().value();
+            var propArray = _.chain(required).zipWith(optional, _.merge).reverse().value();
+            
+            // TODO ideally we could perform this extension data fetch as part of (or at least parallel to)
```

Comment:
Yikes? Let's talk about how we can make this better before merging. This is performance-sensitive.